docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+## Table of Contents
+
+The playbook is divided into three sections:
+
+- **Supporting the New Architecture in your Library**
+  - Prerequisites for Supporting the New Architecture in JavaScript
+  - Enabling the New Architecture in your Library
+    - Android, iOS
+- **Supporting the New Architecture in your App**
+  - Prerequisites for Supporting the New Architecture in your App
+  - Enabling the New NativeModule System (TurboModule) in your App
+    - Android, iOS
+  - Enabling the New Renderer (Fabric) in your App
+    - Android, iOS
+- **Appendix**
+
+# Supporting the New Architecture in Your Library
+
+## Prerequisites for Supporting the New Architecture in JavaScript
+
+The following steps will help ensure your modules and components are ready for the new architecture.
+
+### TurboModules: Define Specs in JavaScript
+
+Under the TurboModule system, the JavaScript spec will serve as the source of truth for the methods that are provided by each native module. Without the JavaScript spec, it is up to you to ensure your public method signatures are equivalent on Android and iOS.
+
+As the first step to adopting the new architecture, you will start by creating these specs for your native modules. You can do this, right now, prior to actually migrating your native module library to the new architecture. Your JavaScript spec will be used later on to generate native interface code for all the supported platforms, as a way to enforce uniform APIs across platforms.
+
+#### Writing the JavaScript Spec
+
+The JavaScript spec defines all APIs that are provided by the native module, along with the types of those constants and functions. You will be using Flow to type the input arguments and outputs of your native module’s methods.
+
+By convention, JavaScript spec files are named `Native<MODULE_NAME>.js` and they export a `TurboModuleRegistry` `Spec` object. The following is a basic JavaScript spec template:
+
+```javascript
+'use strict';
+
+import type {TurboModule} from 'react-native/Libraries/TurboModule/RCTExport';
+import {TurboModuleRegistry} from 'react-native';
+
+export interface Spec extends TurboModule {
+  +getConstants: () => {||};
+
+  // your module methods go here, for example:
+  +getString(id: string): Promise<string>;
+}
+
+export default (TurboModuleRegistry.get<Spec>('<MODULE_NAME>'): ?Spec);
+```
+
+#### Supported Flow Types
+
+You will be using [Flow type annotations](https://flow.org/en/docs/types/) to define your JavaScript spec. Keeping in mind that the goal of defining a JavaScript spec is to ensure the generated native interface code is type safe, the set of supported Flow types will be those that can be mapped one-to-one to a corresponding type on the native platform.
+
+<!-- alex ignore savage -->
```

Comment:
?

Reply:
I had to add a [alex ignore](https://alexjs.com/) here otherwise I would get the following:

```
docs/new-architecture-playbook.md
  66:36-66:45  warning  `primitive` may be insensitive, use `simple`, `indigenous`, `hunter-gatherer` instead  savage  retext-equality
```